contracts/src/access/control.rs

```diff
@@ -119,6 +119,7 @@ pub struct RoleData {
 #[storage]
 pub struct AccessControl {
     /// Role identifier -> Role information.
+    #[allow(clippy::used_underscore_binding)]
```

Comment:
In response to :
> I believe we can apply `#[allow(clippy::used_underscore_binding)]` for the whole contracts library

We would then not get warnings for:

```rust
let _x = 1u32;
let y = _x + 1; // should be a warning
```

Reply:
Fair enough. I am ok with leaving as it is.

Reply:
It would be useful to have an attribute similar to [`[serde(rename = ...)]`](https://serde.rs/field-attrs.html#rename), so we could use Rust conventions for naming fields, and just rename those fields when building the contract :thinking:

Reply:
Actually, are we really required to use the same names as in Solidity?

It is a Solidity convention to prefix private state fields with an underscore `_`, but this is not the case in Rust. 

Why can't we just omit the underscore?

Reply:
@bidzyyys @qalisander

Reply:
@0xNeshi this is the corresponding issue #386 that we have. Tbh I do not like either as it is against Rust attitude. Right now I would merge as it is and then focus on renaming based on our common agreement.

Reply:
Don't like `_*` either

Reply:
Seems the agreement is there :smile: 
Nevertheless, I'd prefer this be done as part of a separate PR linked to the [issue you mentioned](https://github.com/OpenZeppelin/rust-contracts-stylus/issues/386)